client/src/views/EventsView.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+  <v-toolbar class="bg-primaryWhite">
+    <v-text-field
+      prepend-inner-icon="mdi-magnify"
+      hide-details
+      placeholder="Search Events"
+      clearable
+      v-model="searchQuery"
+    />
+    <v-btn
+      v-if="tempIsTeamAdmin"
+      icon="mdi-plus"
+      size="small"
+      id="add-event-btn"
+      class="bg-primaryRed text-primaryWhite"
+      @click="isAddingEvent = true"
+    ></v-btn>
+  </v-toolbar>
+  <div id="cards-container">
+    <br />
+    <Event
+      v-for="(event, index) in filteredEventsList"
+      key="event.id"
+      :id="event.id"
+      :name="event.name"
+      :start-date="event.startDate"
+      :end-date="event.endDate"
+      :description="event.description"
+      :can-edit="tempIsTeamAdmin && event.isPrivate"
+      :is-private="event.isPrivate"
```

Comment:
instead of passing every attribute why not create an event object and just pass the object?